tests/test_sinogram.py

```diff
@@ -135,4 +139,84 @@ def test_multidim(num_ang):
+    rads = angles / 180 * np.pi
+    sinogram = masked_image.project(rads)
+    sinogram_np = sinogram.asnumpy()
+    back_project = sinogram.back_project(rads)
+
+    assert masked_image.shape == back_project.shape, "The shape must be the same."
+
+    # generate circular mask w/ radius 1 to reconstructed image
+    # aim to remove discrepencies for the edges of the image
+    g = grid_2d(back_project.resolution, normalized=True, shifted=True)
+    mask = g["r"] < 1
+    our_back_project = back_project.asnumpy()[0] * mask
+
+    # generating sci-kit image backproject method w/ no filter
+    sk_image_iradon = iradon(sinogram_np[0].T, theta=angles[::-1], filter_name=None)
+
+    # we apply a normalized root mean square error on the images to find relative error to range of ref. image
+    # Note: tolerance is typically < 0.2 regardless of angles, pixels, etc.
+    nrmse = np.sqrt(np.mean((our_back_project - sk_image_iradon) ** 2)) / (
+        np.max(sk_image_iradon - np.min(sk_image_iradon))
+    )
+    assert (
+        nrmse < SK_TOL_BACKPROJECT
+    ), f"NRMSE is too high: {nrmse}, expected less than {SK_TOL_BACKPROJECT}"
+
+
+def test_back_project_multidim(num_ang):
+    """
+    Test Line.backproject on a stack of images. Extension of back_project_single but for multi-dimensional stacks. Similar to forward_multidim test.
+    """
+    L = 512  # pixels
+    n = 3
+    m = 2
+
+    g = grid_2d(L, normalized=True, shifted=True)
+    mask = g["r"] < 1
+
+    # Generate images
+    imgs = Image(np.random.random((m, n, L, L))) * mask
+    angles = np.linspace(0, 360, num_ang, endpoint=False)
+    rads = angles / 180 * np.pi
+
+    # apply a forward project on the image, then backwards
+    ours_forward = imgs.project(rads)
+    ours_backward = ours_forward.back_project(rads)
+
+    # Compare
+    reference_back_projects = np.empty((m, n, L, L))
+    for i in range(m):
+        for j in range(n):
+            img = imgs[i, j]
+            # Compute the singleton case, and compare with stack.
+            single_sinogram = img.project(rads)
+            back_project = single_sinogram.back_project(rads)
+
+            # These should be allclose up to determinism.
+            np.testing.assert_allclose(ours_backward[i, j : j + 1], back_project[0])
+
+            # Next individually compute sk's iradon transform for each image.
+            reference_back_projects[i, j] = iradon(
+                single_sinogram.asnumpy()[0].T, theta=angles[::-1], filter_name=None
+            )
+
+    # apply a mask, then find the NRMSE on the collection of images
+    # similar tolerance level to single project test
+    nrmse = np.sqrt(
+        np.mean((ours_backward.asnumpy() * mask - reference_back_projects) ** 2)
+    ) / (np.max(reference_back_projects) - np.min(reference_back_projects))
+    assert (
+        nrmse < SK_TOL_BACKPROJECT
+    ), f"NRMSE is too high for image ({i},{j}): {nrmse}, expected less than {SK_TOL_BACKPROJECT}"
```

Comment:
This `({i}, {j})` will always be (3, 2) since this message is outside the loop you are referencing.

Reply:
I deleted the `assert` and used a different comparison

<img width="360" alt="Screenshot 2024-08-06 at 12 11 31 PM" src="https://github.com/user-attachments/assets/24ecda92-836b-4a50-85e3-49ec4ce7866f">

Reply:
oops, will change the error message as well